cpp_guidelines/.clang-format

```diff
@@ -0,0 +1,107 @@
+---
+Language: Cpp
+Standard: c++17
+DisableFormat:   false
+AccessModifierOffset: -4
+AlignAfterOpenBracket: Align
```

Comment:
**VOTE**
```
current | AlwaysBreak |
--------|-------------|
 0      |   1         |
```

When changing this to `AlwaysBreak` the arguments of a function (also templates) will moved to the next line and will be indented instead of aligned.
This has big advantages when a function name is refactored and the length changed, in that case the diff is much smaller and will not effect unchanged lines.

# Example:
current setting:
```C++
template <typename T_AtomicAddOp, typename ParticleAssign, int T_begin,
          int T_end>
struct DepositCurrent<T_AtomicAddOp, ParticleAssign, T_begin, T_end, DIM2>
    : public BaseMethods<ParticleAssign>
{
    template <typename T_Cursor, typename T_Acc>
    DINLINE void operator()(T_Acc const& acc, const T_Cursor& cursorJ,
                            const Line<float2_X>& line,
                            const float_X chargeDensity,
                            const float_X velocityZ) const
    {
        using namespace cursor::tools;
        cptCurrent1D(acc, cursorJ, line,
                     cellSize.x() * chargeDensity / DELTA_T);
        cptCurrent1D(
            acc, twistVectorFieldAxes<pmacc::math::CT::Int<1, 0>>(cursorJ),
            rotateOrigin<1, 0>(line), cellSize.y() * chargeDensity / DELTA_T);
        cptCurrentZ(acc, cursorJ, line, velocityZ * chargeDensity);
    }

    /** deposites current in x-direction
     *
     * \param cursorJ cursor pointing at the current density field
     * of the particle's cell \param line trajectory of the virtual
     * particle \param currentSurfaceDensity surface density
     */
    template <typename CursorJ, typename T_Line, typename T_Acc>
    DINLINE void cptCurrent1D(T_Acc const& acc, CursorJ cursorJ,
                              const T_Line& line,
                              const float_X currentSurfaceDensity) const
    {
```

Suggestion:
```C++
template <
    typename T_AtomicAddOp, typename ParticleAssign, int T_begin, int T_end>
struct DepositCurrent<T_AtomicAddOp, ParticleAssign, T_begin, T_end, DIM2>
    : public BaseMethods<ParticleAssign>
{
    template <typename T_Cursor, typename T_Acc>
    DINLINE void operator()(
        T_Acc const& acc, const T_Cursor& cursorJ, const Line<float2_X>& line,
        const float_X chargeDensity, const float_X velocityZ) const
    {
        using namespace cursor::tools;
        cptCurrent1D(
            acc, cursorJ, line, cellSize.x() * chargeDensity / DELTA_T);
        cptCurrent1D(
            acc, twistVectorFieldAxes<pmacc::math::CT::Int<1, 0>>(cursorJ),
            rotateOrigin<1, 0>(line), cellSize.y() * chargeDensity / DELTA_T);
        cptCurrentZ(acc, cursorJ, line, velocityZ * chargeDensity);
    }

    /** deposites current in x-direction
     *
     * \param cursorJ cursor pointing at the current density field
     * of the particle's cell \param line trajectory of the virtual
     * particle \param currentSurfaceDensity surface density
     */
    template <typename CursorJ, typename T_Line, typename T_Acc>
    DINLINE void cptCurrent1D(
        T_Acc const& acc, CursorJ cursorJ, const T_Line& line,
        const float_X currentSurfaceDensity) const
    {
```
```

Reply:
current | AlwaysBreak |
--------|-------------|
 0      |   2         |

Reply:
current | AlwaysBreak |
--------|-------------|
 0      |   3         |

Reply:
current | AlwaysBreak | DontAlign |
--------|-------------|-------------|
 0      |   2        | 1         |

Reply:
current | AlwaysBreak |DontAlign|
--------|-------------|-----|
 0      |   4         |1|

Reply:
current | AlwaysBreak | DontAlign
-- | -- | --
0 | 5 | 1

Reply:
current | AlwaysBreak | DontAlign |
--------|-----------|---------------|
 1      |   5 |  1

Reply:
current | AlwaysBreak | DontAlign
-- | -- | --
1 | 6 | 1

Reply:
Reminder: Voting will be closed on 2020-09-26 at 11:13 HZDR time.

Reply:
current | AlwaysBreak | DontAlign
--------|-----------|------|
 2      |   6 | 1 |

Reply:
Voting closed, the winner is `AlwaysBreak`.